src/main/java/controllers/CalculatorController.java

```diff
@@ -0,0 +1,110 @@
+package controllers;
+
+import model.CalculatorModel;
+import util.Constants;
+import view.CalculatorView;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.lang.reflect.InvocationTargetException;
+
+public class CalculatorController implements PropertyChangeListener {
+
+    private final CalculatorView view;
+    private final CalculatorModel model;
+
+    public CalculatorController(CalculatorView view, CalculatorModel model) {
+        this.view = view;
+        this.model = model;
+
+        model.addPropertyChangeListener(this);
+
+        addListeners();
+        addExtensibleFunctionNames();
+    }
+
+    private void addListeners() {
+        view.addClearButtonListener(l -> model.resetEverything());
+
+        view.addDeleteButtonListener(l -> model.deleteFromBigText());
+
+        view.addDecimalPointButtonListener(l-> model.addDecimalPointToBigText());
+
+        view.addDirectOperationButtonListener(l -> {
+            String operation = getClickedButtonText(l);
+            model.setOperation(operation);
+            model.setDirect(true);
+            try {
+                model.calculate(false);
+            } catch (InvocationTargetException | NoSuchMethodException | InstantiationException |
+                     IllegalAccessException e) {
+                view.displayErrorMessage("Ups");
```

Comment:
Ich denke mal, das "Ups" ist absichtlich da?

Reply:
die error message sollte eigentlich überhaupt nicht auftauchen können, da ja in dem dropdown nur die sachen über die reflections angezeigt werden, und man dann ja auch nur die funktionen auswählen kann. Aber ich überlege mir da mal was nutzerfreundlicheres